grid/chat/workspaces/maps/discord.suma

```diff
@@ -0,0 +1,180 @@
+profile = "chat/workspaces@1.0"
+provider = "discord"
+
+"""
+GetWorkspaces map
+"""
+map GetWorkspaces {
+  http GET "/users/@me/guilds" {
+    request "application/json" {
+      headers {
+        "Authorization" = `${input.asBot ? 'Bot' : 'Bearer'} ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      workspaces = call MapWorkspaces(workspaces = body)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+
+      map result {
+        workspaces = workspaces
+        rateLimit = rateLimit
+      }
+    }
+
+    response {
+      error = call MapDiscordError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+operation MapWorkspaces {
+  workspaces = args.workspaces.map(workspace => {
+    /**
+     * Returns UNIX timestamp from specified discord snowflake id
+     * More about snowflakes: https://discord.com/developers/docs/reference#snowflakes
+     */
+    const discordEpoch = 1420070400000;
+    const getTimestamp = (snowflake) => Number(BigInt(snowflake) >> BigInt(22)) + discordEpoch;
+
+    const resultWorkspace = {
+      id: workspace.id,
+      createdAt: getTimestamp(workspace.id),
+      name: workspace.name,
+    };
+
+    if (workspace.member_count) {
+      resultWorkspace.membersCount = workspace.member_count;
+    }
+
+    if (workspace.icon) {
+      resultWorkspace.icon = workspace.icon;
+    }
+
+    if (workspace.description) {
+      resultWorkspace.description = workspace.description;
+    }
+
+    return resultWorkspace;
+  });
+
+  return workspaces
+}
+
+
+"""
+Operation MapRateLimitDetails
+Returns object mapping rate limit details specified in received headers
+"""
+operation MapRateLimitDetails {
+  headers = args.headers
+  details = {}
+
+  bucket = headers['x-ratelimit-bucket']
+  totalRequests = parseInt(headers['x-ratelimit-limit'])
+  remainingRequests = parseInt(headers['x-ratelimit-remaining'])
+  resetTimestamp = parseFloat(headers['x-ratelimit-reset'])
+  resetAfter = parseInt(headers['x-ratelimit-reset-after'])
+  retryAfter = parseInt(headers['Retry-After'])
+
+  set if (bucket) {
+    details.bucket = bucket
+  }
+
+  set if (!isNaN(totalRequests)) {
+    details.totalRequests = totalRequests
+  }
+
+  set if (!isNaN(remainingRequests)) {
+    details.remainingRequests = remainingRequests
+  }
+
+  set if (!isNaN(resetAfter)) {
+    details.resetAfter = resetAfter
+  }
+
+  set if (!isNaN(resetTimestamp)) {
+    details.resetTimestamp = resetTimestamp * 1000
+  }
+
+  set if (!isNaN(retryAfter)) {
+    details.retryAfter = retryAfter
+  }
+
+  set if (details.remainingRequests && details.totalRequests) {
```

Comment:
`remainingRequestsPercentage` is missing in the outcome because  `remainingRequests` is 0.
```suggestion
  set if (details.remainingRequests !== undefined && details.totalRequests) {
```